### PR TITLE
Components for the top navigation bar

### PR DESCRIPTION
**Pull request by econdepe**
In order to have consistency across `contacts`, `settings`, and eventually `mail`, I created some components here that can be shared across later.

Some CSS classes have been also fixed thanks to @nico3333fr.

### CODE REVIEW

File: components/link/TopNavbarLink.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        <Link to={to} {...rest}>
+            {iconComponent}
+            <span className="navigation-title topnav-linkText">{text}</span>
+        </Link>
+    );
+};
+
+TopNavbarLink.propTypes = {
+    to: PropTypes.string.isRequired,
+    icon: PropTypes.string.isRequired,
```
Comment (EpokK) [inline components/link/TopNavbarLink.js]: Why not using `node` here? I don't understand why we need to handle node and string with `iconComponent`.

File: containers/search/Searchbox.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
+import React from 'react';
+import { c } from 'ttag';
+import PropTypes from 'prop-types';
+import { SearchInput, Icon } from 'react-components';
+
+import { classnames } from '../../helpers/component';
+
+const Searchbox = ({ className = '', placeholder = '', value = '', onSearch }) => {
+    return (
+        <div className={classnames(['searchbox-container relative flex-item-centered-vert', className])}>
```
Comment (EpokK) [inline containers/search/Searchbox.js]: Where do you define the "loupe" icon?

File: components/link/TopNavbarLink.js (hunk 1 above)
Comment (econdepe) [reply components/link/TopNavbarLink.js]: I just wanted to imitate what we have in the `Button` component, where `icon` can be either a string (name of the icon) or a node (icon itself). But I forgot to modify the PropType accordingly. I did that with the last push.

File: containers/search/Searchbox.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                    onChange={onSearch}
+                    id="global_search"
+                    placeholder={placeholder}
+                    className="searchbox-field"
+                />
+            </label>
+            <button type="button" className="searchbox-search-button flex">
+                <Icon name="search" className="fill-white mauto searchbox-search-button-icon" />
+                <span className="sr-only">{c('Action').t`Search`}</span>
+            </button>
```
Comment (EpokK) [inline containers/search/Searchbox.js]: You should add also support for an `advanced` button.
Capture d’écran 2019-10-07 à 10 49 25

File: containers/search/Searchbox.js (hunk 2 above)
Comment (EpokK) [inline containers/search/Searchbox.js]: Can we use a `form` here? And then use `onSubmit={onSearch}` and `type=submit` on `searchbox-search-button`.

File: components/link/TopNavbarLink.js (hunk 1 above)
Comment (EpokK) [reply components/link/TopNavbarLink.js]: I don't see the point of using both, just use `PropTypes.string` please.

**Pull request merged**